modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/fork/ForkInfoStorage.scala

```diff
@@ -0,0 +1,35 @@
+package org.tessellation.sdk.infrastructure.fork
+
+import cats.Monad
+import cats.effect.Ref
+import cats.syntax.functor._
+
+import org.tessellation.schema.peer.PeerId
+import org.tessellation.sdk.config.types.ForkInfoStorageConfig
+import org.tessellation.sdk.domain.fork.{ForkInfo, ForkInfoQueue, ForkInfoStorage}
+
+import eu.timepit.refined.auto._
+
+object ForkInfoStorage {
+
+  type ForkInfoMap = Map[PeerId, ForkInfoQueue]
+
+  def make[F[_]: Monad: Ref.Make](config: ForkInfoStorageConfig): F[ForkInfoStorage[F]] =
+    Ref[F]
+      .of(Map.empty[PeerId, ForkInfoQueue])
+      .map(make(_, config))
+
+  def make[F[_]: Monad](
+    storeRef: Ref[F, ForkInfoMap],
+    config: ForkInfoStorageConfig
+  ): ForkInfoStorage[F] = new ForkInfoStorage[F] {
+    def add(peerId: PeerId, entry: ForkInfo): F[Unit] = storeRef.update { s =>
+      val updated = s.getOrElse(peerId, ForkInfoQueue.empty(config.maxSize)).add(entry)
+
+      s + (peerId -> updated)
+    }
+
```

Comment:
It is probably faster and cleaner to use `Map.updatedWith`

Reply:
I made an implementation with `updatedWith` and spoke with @scasplte2 about it. We ultimately decided to revert to this.